webpack.config.js

```diff
@@ -0,0 +1,54 @@
+const path = require("path");
+const HtmlWebpackPlugin = require("html-webpack-plugin");
+const MiniCssExtractPlugin = require("mini-css-extract-plugin");
+
+const config = {
+  entry: "./src/app/App.js",
+  output: {
+    path: path.resolve(__dirname, "dist"),
+    filename: "App.js",
+  },
+
+  mode: "development",
+  devServer: {
+    static: {
+      directory: path.join(__dirname, "public"),
+    },
+    compress: true,
+    port: 9000,
+  },
+  module: {
+    rules: [
+      {
+        test: /\.m?js$/,
+        exclude: /node_modules/,
+        use: {
+          loader: "babel-loader",
+          options: {
+            presets: [["@babel/preset-env", { targets: "defaults" }]],
+          },
+        },
+      },
+      {
+        test: /\.css$/,
+        use: [MiniCssExtractPlugin.loader, "css-loader"],
+      },
+      {
+        test: /\.scss$/,
+        use: [MiniCssExtractPlugin.loader, "css-loader", "sass-loader"],
+      },
+    ],
+  },
+  plugins: [
+    new HtmlWebpackPlugin({
+      templateContent: ({ htmlWebpackPlugin }) =>
+        '<!DOCTYPE html><html><head><meta charset="utf-8"><title>' +
+        "Battleship" +
+        '</title></head><body><div id="app"></div></body></html>',
+      filename: "index.html",
+    }),
+    new MiniCssExtractPlugin(),
```

Comment:
Is it needed if it is empty?

https://webpack.js.org/plugins/mini-css-extract-plugin/